packages/nativescript-audio-player/index.ios.ts

```diff
@@ -0,0 +1,377 @@
+                } else {
+                  reject();
+                }
+              });
+
+              this._task.resume();
+            } catch (ex) {
+              console.error(ex);
+              if (this.errorCallback) {
+                this.errorCallback({ ex });
+              }
+              reject(ex);
+            }
+          }
+        }
+
+        // resolve(true);
+      } catch (ex) {
+        reject(ex);
+      }
+    });
+  }
+
+  public pause(): Promise<any> {
+    return new Promise((resolve, reject) => {
+      try {
+        if (this._player && this._player.playing) {
+          this._player.pause();
+        }
+        resolve(true);
+      } catch (ex) {
+        if (this.errorCallback) {
+          this.errorCallback({ ex });
+        }
+        reject(ex);
+      }
+    });
+  }
+
+  public play(): Promise<any> {
+    return new Promise((resolve, reject) => {
+      try {
+        if (!this.ready) {
+          reject('Audio source not ready, call prepareAudio(options) first!');
+          return false;
+        }
+        if (!this.isAudioPlaying()) {
+          this._player.play();
+        }
+        resolve(true);
+      } catch (ex) {
+        if (this.errorCallback) {
+          this.errorCallback({ ex });
+        }
+        reject(ex);
+      }
+    });
+  }
+
+  public resume(): void {
+    if (this._player) {
+      this._player.play();
+    }
+  }
+
+  public playAtTime(time: number): void {
+    if (this._player) {
+      this._player.playAtTime(time);
+    }
+  }
+
+  public seekTo(time: number): Promise<any> {
+    return new Promise((resolve, reject) => {
+      try {
+        if (this._player) {
+          this._player.currentTime = time;
+        }
+        resolve(true);
+      } catch (ex) {
+        reject(ex);
+      }
+    });
+  }
+
+  public dispose(): Promise<any> {
+    return new Promise((resolve, reject) => {
+      try {
+        if (this._player && this.isAudioPlaying()) {
+          this._player.stop();
+        }
+        const audioSession = AVAudioSession.sharedInstance();
+        audioSession.setActiveError(false);
+        this._reset();
+        resolve(null);
+      } catch (ex) {
+        if (this.errorCallback) {
+          this.errorCallback({ ex });
+        }
+        reject(ex);
+      }
+    });
+  }
+
+  public isAudioPlaying(): boolean {
+    return this._player ? this._player.playing : false;
+  }
+
+  public getAudioTrackDuration(): Promise<string> {
+    return new Promise((resolve, reject) => {
+      try {
+        const duration = this._player ? this._player.duration : 0;
+        resolve(duration.toString());
+      } catch (ex) {
+        if (this.errorCallback) {
+          this.errorCallback({ ex });
+        }
+        reject(ex);
+      }
+    });
+  }
+
+  public changePlayerSpeed(speed) {
+    if (this._player && speed) {
+      // make sure speed is a number/float
+      if (typeof speed === 'string') {
+        speed = parseFloat(speed);
```

Comment:
@drangelod do we need to check for `NaN` after `parseFloat` to make sure that it indeed type of number before setting `this._player.rate`?

Reply:
Can't hurt, I'll add that later today